src/components/ApplicationContainer/index.tsx

```diff
@@ -1,24 +1,45 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Header, navConfig } from "@nypl/dgx-header-component";
 import Footer from "@nypl/dgx-react-footer";
 import Banner from "../Banner";
+import ApiErrors from "../ApiErrors";
+import useFormDataContext from "../../context/FormDataContext";
 
-const ApplicationContainer: React.FC = ({ children }) => (
-  <>
-    <Header skipNav={{ target: "main-content" }} navData={navConfig.current} />
-    <div className="nypl-library-card-app layout-container nypl-ds">
-      <main id="main-content" className="main main--with-sidebar">
-        <div className="content-header">
-          <Banner />
-        </div>
-        <div className="content-primary content-primary--with-sidebar-right">
-          {children}
-        </div>
-        <div className="content-secondary content-secondary--with-sidebar-right" />
-      </main>
-    </div>
-    <Footer />
-  </>
-);
+const ApplicationContainer = ({ children, problemDetail }) => {
+  const errorSection = React.createRef<HTMLDivElement>();
+  const { state } = useFormDataContext();
+  const { errorObj } = state;
+  const errorToDisplay = problemDetail ? problemDetail : errorObj;
+
+  // If there are errors, focus on the element that displays those errors,
+  // for client-side rendering.
+  useEffect(() => {
+    if (errorToDisplay) {
+      errorSection.current.focus();
+    }
+  }, [errorToDisplay]);
```

Comment:
Why is this an array?  It looks like there would only ever be one at a time...no?

Reply:
Yes, there would only ever be one error object (problem detail) at a time, but it can get updated on subsequent requests only on the client-side. If you make a submission (on the client-side) and find out that the username is not available, you'll get an error back and the `errorSection` is focused. If the username is updated and the form is submitted but there was another error, then `errorToDisplay` is now a new error object.

In `useEffect`, the second argument is a dependency array. This array is telling `useEffect` to run every time that `errorToDisplay` gets updated. Every time there is a new error, the error DOM element gets focused.